eth/protocols/eth/handlers.go

```diff
@@ -287,30 +287,59 @@ func handleNewBlockhashes(backend Backend, msg Decoder, peer *Peer) error {
 	return backend.Handle(peer, ann)
 }
 
+type BlockPacket interface {
+	GetBlock() *types.Block
+	SanityCheck() error
+	Name() string
+	Kind() byte
+}
+
+// Implement the GetBlock method for both packet types
+func (p *NewBlockPacket) GetBlock() *types.Block {
+	return p.Block
+}
+
+func (p *NewBlockWithBlobPacket) GetBlock() *types.Block {
+	return p.Block
+}
+
 func handleNewBlock(backend Backend, msg Decoder, peer *Peer) error {
-	// Retrieve and decode the propagated block
-	ann := new(NewBlockPacket)
-	if err := msg.Decode(ann); err != nil {
-		return fmt.Errorf("%w: message %v: %v", errDecode, msg, err)
+	var packet BlockPacket
+
+	// First, try to decode as NewBlockPacket
+	nbp := new(NewBlockPacket)
```

Comment:
Maybe we could add some UTs to check the decode is ok for the old msg and the new msg.